src/filter/atomic_bitvec.rs

```diff
@@ -0,0 +1,355 @@
+use std::{sync::atomic::*, fmt::Display};
+
+/// Bit vector with atomic operations on its bits
+pub(crate) struct AtomicBitVec {
+    data: Box<[AtomicU64]>,
+    bits_count: usize
+}
+
+#[derive(Debug)]
+pub(crate) enum AtomicBitVecError {
+    BitsCountMismatch,
+    DataLengthLessThanRequired
+}
+
+/// Contains helper functions for bit access
+pub(crate) struct OffsetAndMaskCalculator<TItem: Sized> {
```

Comment:
Is generic really needed? The only place it's being used is with `u8`, why not make it `ByteOffsetAndMaskCalculator`?

Reply:
Removed generic parameter and added `_u8` suffix to the function names